docs/manual/shared_memory.rst

```diff
@@ -60,78 +136,181 @@ Save the following xml as cyclonedds.xml.
+
+* Deadline
+
+  * DDS_INFINITY
+
+* Reliability
+
+  * DDS_RELIABILITY_RELIABLE
+
+* Durability
+
+  * DDS_DURABILITY_VOLATILE
+
+* History
+
+  * DDS_HISTORY_KEEP_LAST
+
+  * with depth no larger than the publisher history capacity as set in the configuration file
 
-You can compare the result between native Cyclone DDS and Cyclone DDS with shared memory.
+Whereas for readers, the following QoS settings are prerequisites for shared memory exchange:
 
-***********
-Performance
-***********
+* Liveliness
 
-A performance improvement can be observed for sufficiently large sample sizes and data transfer on the same machine (otherwise the regular network transmission is used and their will be no performance gain).
+  * DDS_LIVELINESS_AUTOMATIC
 
-Once the loan interface is fully implemented transmission speed will be independent of the sample size when shared memory is used, i.e. take constant time.
+* Deadline
 
-*************
-To developers
-*************
+  * DDS_INFINITY
+
+* Reliability
+
+  * DDS_RELIABILITY_RELIABLE
+
+* Durability
+
+  * DDS_DURABILITY_VOLATILE
+
+* History
+
+  * DDS_HISTORY_KEEP_LAST
+
+  * with depth no larger than the subscriber history request as set in the configuration file
+
+If any of these prerequisites are not satisfied, shared memory exchange will be disabled and data transfer will fall back to the network interface.
+
+A further limitation is the maximum number of subscriptions per process for the iceoryx service, which is 127.
+
+Lastly, there is the limit on the operating system. Iceoryx currently has no functioning implementation for the Windows operating system, this is `under development <https://github.com/eclipse/iceoryx/issues/33>`_.
+
+Loan Mechanism
+--------------
+
+If the choice to use shared memory exchange is made, additional performance gains can be made by using the loan mechanism on the writer side.
+The loan mechanism directly allocates memory from the iceoryx shared memory pool, and provides this to the user in the shape of the message data type.
+Thereby eliminating a copy step in the publication process.
+
+.. code-block:: C
+
+  message_type *loaned_sample;
+  dds_return_t status = dds_loan_sample(writer, (void**)&loaned_sample);
+
+If *status* returns **DDS_RETCODE_OK**, then *loaned_sample* will contain a pointer to the memory pool object, in all other cases, *loaned_sample* should not be dereferenced.
+For requesting loaned samples, the writer used to request the loaned sample should be of the same data type as the sample that you are writing in it, since necessary information about the data type is supplied by the writer.
+
+The user is limited in this case by the maximum number of outstanding loans, defined by **MAX_PUB_LOANS** (default set to 8). This is the maximum number of loaned samples that can be outstanding process wide from the shared memory, before some must be returned (handed back to the publisher through *dds_write*) before requesting new loaned samples.
```

Comment:
I think this limit is not process wide but per publisher, as the constant name suggests.

Reply:
After looking in the code: you are correct.

Reply:
Maybe I missed something, but it still says process wide. Shouldn't  it be per writer?